eth/stagedsync/stage_headers.go

```diff
@@ -488,9 +488,8 @@ func handleNewPayload(
 	if parent == nil {
 		log.Info(fmt.Sprintf("[%s] New payload missing parent", s.LogPrefix()))
 		hashToDownload := header.ParentHash
-		heightToDownload := headerNumber - 1
 		cfg.hd.SetPoSDownloaderTip(headerHash)
-		schedulePoSDownload(requestId, hashToDownload, heightToDownload, s, cfg)
+		schedulePoSDownload(requestId, hashToDownload, headerNumber, s, cfg)
```

Comment:
The names of the arguments to `schedulePoSDownload` are `hashToDownload`, `heightToDownload`. So the implication is that they refer to the same block, which was previously the case:
```
hashToDownload := header.ParentHash
heightToDownload := headerNumber - 1
```
But with this change they refer to different things (header's number and its _parent's_ hash). So please either revert this change (not the entire PR, just the `schedulePoSDownload` parameters part) or rename the params of `schedulePoSDownload`.